### DISCUSSION TITLE
Inaccurate tagging of csubjpass

### DISCUSSION THREAD

**Discussion by fatihbozdag** [explosion/spaCy#11431] [Help: Model Advice] [answered]
Greetings all,

I am working on extracting subordinate clauses with Spacy. However, dependency parsing provides incorrect results for `csubjpass` if there is another word prior to passive verb. 

For instance: 
```
doc_1 = nlp('what she said was well received')
for t in doc_1:
    print(t.text, t.dep_, t.head.text)

what dobj said
she nsubj said
said csubj was
was ROOT was
well advmod received
received acomp was
```
However, 

```
doc_2 = nlp('what she said was received')
for t in doc_1:
    print(t.text, t.dep_, t.head.text)

what dobj said
she nsubj said
said csubjpass received
was auxpass received
received ROOT received
```
In first case, `said` should have been tagged as `csubjpass`,  as exemplified [here:](https://universaldependencies.org/u/overview/complex-syntax.html#subordination) 

How should I proceed, then? Any suggestions?

```
Spacy ver: 3.4
Spacy Modal: en_core_web_trf
```

**Comment by polm** [comment, marked as answer]
If this issue is systematic like you describe, I would recommend writing a small custom component to fix the annotations. You could also probably use the [AttributeRuler](https://spacy.io/api/attributeruler). 

In addition to #3052 about errors in general, I suspect the model has trouble with this because this kind of usage isn't very common in the training data (or in many kinds of writing, for that matter).

**Reply by fatihbozdag** [reply]
Well, strange thing is that Stanza and UD Models also suffer the same issue.  I'd better start working on AttributeRuler.
